src/test/scala/io/iohk/ethereum/cli/CliCommandsSpec.scala

```diff
@@ -0,0 +1,99 @@
+package io.iohk.ethereum.cli
+
+import org.scalatest.EitherValues
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+class CliCommandsSpec extends AnyFlatSpec with Matchers with EitherValues {
+
+  import CliCommands._
+  import Fixture._
+
+  behavior of generatePrivateKeyCommand
+  it should "generate correct private key" in {
+    api.parse(Seq(generatePrivateKeyCommand)) shouldBe a[Right[_, _]]
+  }
+
+  behavior of deriveAddressCommand
+  it should "derive address from private key" in {
+    api.parse(Seq(deriveAddressCommand, privateKey)).right.value shouldBe address
+  }
+
+  it should "return an error when called without private key" in {
+    api.parse(Seq(deriveAddressCommand)) shouldBe a[Left[_, _]]
+  }
+
+  behavior of generateAllocsCommand
+  it should "generate correct alloc using private key" in {
+    api
+      .parse(
+        Seq(
+          generateAllocsCommand,
+          argument(keyOption, Some(privateKey)),
+          argument(balanceOption, Some(requestedBalance))
+        )
+      )
+      .right
+      .value shouldBe s""""alloc": {$address: { "balance": $requestedBalance }}"""
+  }
+
+  it should "generate more than one alloc" in {
+    api
+      .parse(
+        Seq(
+          generateAllocsCommand,
+          argument(keyOption, Some(privateKey)),
+          argument(keyOption, Some(privateKey2)),
+          argument(balanceOption, Some(requestedBalance))
+        )
+      )
+      .right
+      .value shouldBe s""""alloc": {$address: { "balance": $requestedBalance }, $address2: { "balance": $requestedBalance }}"""
+  }
+
+  it should "generate allocs using addresses" in {
+    api
+      .parse(
+        Seq(
+          generateAllocsCommand,
+          argument(addressOption, Some(address)),
+          argument(addressOption, Some(address2)),
+          argument(balanceOption, Some(requestedBalance))
+        )
+      )
+      .right
+      .value shouldBe s""""alloc": {$address: { "balance": $requestedBalance }, $address2: { "balance": $requestedBalance }}"""
+  }
+
+  it should "generate allocs using both keys and addresses" in {
+    api
+      .parse(
+        Seq(
+          generateAllocsCommand,
+          argument(addressOption, Some(address)),
+          argument(keyOption, Some(privateKey2)),
+          argument(addressOption, Some(address3)),
+          argument(balanceOption, Some(requestedBalance))
+        )
+      )
+      .right
+      .value shouldBe s""""alloc": {$address: { "balance": $requestedBalance }, $address3: { "balance": $requestedBalance }, $address2: { "balance": $requestedBalance }}"""
+  }
+
+}
+
+object Fixture {
+
+  def argument(name: String, value: Option[Any] = None): String = s"--$name${value.fold("")(v => s"=${v.toString}")}"
```

Comment:
very minor in current state - that `Any` there, it's fine now, but I'd consider using just `String` instead so any conversion needs to happen explicitly